x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/simulation_state_machine/samples_fetcher_actor.ts

```diff
@@ -9,19 +9,22 @@ import { i18n } from '@kbn/i18n';
 import { Condition, SampleDocument } from '@kbn/streams-schema';
 import { fromPromise, ErrorActorEvent } from 'xstate5';
 import type { errors as esErrors } from '@elastic/elasticsearch';
-import { DateRangeContext } from '../../../../../state_management/date_range_state_machine';
+import { firstValueFrom } from 'rxjs';
 import { SimulationMachineDeps } from './types';
 
 export interface SamplesFetchInput {
   condition?: Condition;
   streamName: string;
-  absoluteTimeRange: DateRangeContext['absoluteTimeRange'];
 }
 
 export function createSamplesFetchActor({
   streamsRepositoryClient,
-}: Pick<SimulationMachineDeps, 'streamsRepositoryClient'>) {
+  timeState$,
+}: Pick<SimulationMachineDeps, 'streamsRepositoryClient' | 'timeState$'>) {
   return fromPromise<SampleDocument[], SamplesFetchInput>(async ({ input, signal }) => {
+    const {
+      timeState: { asAbsoluteTimeRange },
+    } = await firstValueFrom(timeState$);
```

Comment:
We wouldn't need this await if `timeState$` would be a BehaviorSubject, but not sure whether we need to change it